### ISSUE TITLE
Slow lagging behaviour for certain large C struct-type 

### ISSUE THREAD

**Issue by RolfNoot** [microsoft/vscode-cpptools#5571]
**Type: LanguageService**
Extreme lagging at mouse-hover over variables for large struct type.

**Describe the bug**
- OS and Version: OSX 10.14.6 / Darwin x64 18.7.0
- VS Code Version: 1.45.1
- C/C++ Extension Version: 0.28.1

When hovering over the variables, the text editor freezes for about 4 seconds (no typing / scrolling possible). Only for this specific struct type, other variables in the same file are not resulting in the issue.

**Comment by michelleangela** [comment]
@RolfNoot 
Can you provide the following log that could show information about what's happening with the language server when hover is done over the struct type:

- Log output from enabling the [language server](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)

**Comment by RolfNoot** [comment]
@michelleangela 
I had set "files.associations" in settings for the .c file to "txt" to get rid of the lagging. Strange thing is that I do not see the issue right now, after setting it back to .c. I've made changes the structure and the issue magically disappeared. I am pretty sure the old structure caused the problem, even after a restart of the system, the issue persisted.

Will revert to the old structure to see if the issue appears again. Keep you updated.

Thanks,
Rolf
